torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp

```diff
@@ -335,6 +336,7 @@ static void PrepareForRemoveMutations(MutationRemover& mr, Block* b) {
         TORCH_INTERNAL_ASSERT(nullptr != newNode);
         node->replaceInput(index, newNode->output());
         input->replaceAllUsesAfterNodeWith(node, newNode->output());
+        goto restart;
```

Comment:
Yuck. Can't we do this with say do-while loop:
```
  bool needs_restart = false;
  do {
      needs_restart = false;
      ....
      if (cond_met) needs_restart = true;
  } while (needs_restart);

Reply:
And I love that `clang-tidy` actually complained about that one :)

Reply:
I knew I'd get complaints about this but I had to try :-p.  IMO it's just as readable as the boolean but whatever.